ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -145,7 +145,18 @@ private OperationController getOperationController() {
    */
   @Override
   public Future<ReadableStreamChannel> getBlob(String blobId) {
-    return getBlob(blobId, null);
+    return getBlob(blobId, null, null);
+  }
+
+  /**
+   * Requests for blob data asynchronously and returns a future that will eventually contain a
+   * {@link ReadableStreamChannel} that represents blob data on a successful response.
+   * @param blobId The ID of the blob for which blob data is requested.
+   * @param options The options associated with the request.
+   * @return A future that would contain a {@link ReadableStreamChannel} that represents the blob data eventually.
+   */
+  public Future<ReadableStreamChannel> getBlob(String blobId, GetBlobOptions options) {
```

Comment:
No, it's was removed from the interface, but I forgot to remove these methods from the implementation.  I will clean up these extra methods (and change the `getBlob(String)` api)